CHANGELOG.md

```diff
@@ -1,3 +1,1038 @@
+`mydomain/#!/a/b/c`.
+
+If you actually wanted to have no hash-prefix then you should configure
+this by adding a configuration block to you application:
+
+```
+appModule.config(['$locationProvider', function($locationProvider) {
+  $locationProvider.hashPrefix(&quot;&quot;);
+}]);
+```
+- **fix(select): don't register options when select has no ngModel ([e8c2e1](https://github.com/angular/angular.js/commit/e8c2e119758e58e18fe43932d09a8ff9f506aa9d))**:
+
+Option elements will no longer set their value attribute from their text value when their select
+element has no ngModel associated. Setting the value is only needed for the select directive to
+match model values and options. If no ngModel is present, the select directive doesn't need it.
+
+This should not affect many applications as the behavior was undocumented and not part of a public
+API. It also has no effect on the usual HTML5 behavior that sets the select value to the option text
+if the option does not provide a value attribute.
+
+- **fix($resource): add semicolon to whitelist of delimiters to unencode in URL params ([2456ab](https://github.com/angular/angular.js/commit/2456ab63a613902d21c151445f9c697a76ab43b3))**:
+
+Although it shouldn't matter in practice (since both the encoded and the unencoded `;` character would
+be interpreted identically by the server), this change could break some tests: For example, where
+`$httpBackend` was set up to expect an encoded `;` character, but the request is made to the URL with an
+unencoded `;` character.
+
+- **feat(input): add support for binding to `input[type=range]` ([913016](https://github.com/angular/angular.js/commit/9130166767c4792c5d32d08a918fc7becf32c9a6))**:
+
+Due to the way that `input[type=range]` elements behave this feature modifies the behavior of such elements
+when bound to `ngModel`:
+
+- Like input[number], it requires the model to be a Number, and will set the model to a Number
+- it supports setting the min/max values via the min/max and ngMin/ngMax attributes
+- it follows the browser behavior of never allowing an invalid value. That means, when the browser
+converts an invalid value (empty: null, undefined, false ..., out of bounds: greater than max, less than min)
+to a valid value, the input will in turn set the model to this new valid value via $setViewValue.
+  - this means a range input will never be required and never have a non-Number model value, once the
+    ngModel directive is initialized.
+  - this behavior is supported when the model changes and when the min/max attributes change in a way
+    that prompts the browser to update the input value.
+  - ngMin / ngMax do not prompt the browser to update the values, as they don't set the attribute values.
+    Instead, they will set the min / max errors when appropriate
+- browsers that do not support input[range] (IE9) handle the input like a number input (with validation etc.)
```

Comment:
_input[range]_ --> _\`input[range]\`_ (or else `[number]` is interpreted as link)